<!-- This launch file is used to calibrate the relative position and orientation of camera frame wrt. mdworld -->
<!-- Reference: https://github.com/IFL-CAMP/easy_handeye -->

<!-- Original of this launch file is from mico_perception/ easy_handeye/ mico_kinect_calibration -->


<launch>
    <!--arg name="namespace_prefix" default="ur5_kinect_handeyecalibration" /-->


    <arg name="marker_size" default="0.100" doc="Size of the ArUco marker used, in meters" />
    <arg name="marker_id" default="566" doc="The ID of the ArUco marker used" />

    <!-- start the Kinect -->
    <!--include file="$(find freenect_launch)/launch/freenect.launch" >
        <arg name="depth_registration" value="true" />
    </include-->

    <!-- start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="/kinect2/qhd/camera_info" />
        <remap from="/image" to="/kinect2/qhd/image_color_rect" />
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="reference_frame"    value=""/>
        <param name="camera_frame"       value="kinect2_rgb_optical_frame"/>
        <param name="marker_frame"       value="marker_frame" />
    </node>

    <!-- start the robot -->
    <!--include file="$(find ur_bringup)/launch/ur5_bringup.launch">
        <arg name="limited" value="true" />
        <arg name="robot_ip" value="192.168.0.21" />
    </include>
    <include file="$(find ur5_moveit_config)/launch/ur5_moveit_planning_execution.launch">
        <arg name="limited" value="true" />
    </include-->

    <!-- start easy_handeye -->
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="eye_on_hand" value="false" />

        <arg name="tracking_base_frame" value="kinect2_rgb_optical_frame" />
        <arg name="tracking_marker_frame" value="marker_frame" />
        <arg name="robot_base_frame" value="world" />
        <arg name="robot_effector_frame" value="mico_link_hand" />

        <arg name="freehand_robot_movement" value="false" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />
    </include>

</launch>
